src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -2148,7 +2156,7 @@ private void StartNestedResourceInfo(ODataJsonLightReaderNestedResourceInfo read
             odataUri.Path = odataPath;
 
             this.EnterScope(new JsonLightNestedResourceInfoScope(readerNestedResourceInfo, navigationSource,
-                targetResourceType, odataUri));
+                targetResourceType, odataUri, resourceNullable));

```

Comment:
Can we get the "targetResourceTypeReference"?

Reply:
I believe it can: See https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/JsonLight/ODataJsonLightResourceDeserializer.cs#L1193

So, the structuredPropertyTypeOrItemType is from the property type. the property type is type reference. I do think you should use "structuredProperty.Type" directly.

Reply:
This is on the serializer. We don't have access to the property on the reader. Again will check with you in person.

Reply:
The get method for the nestedresourcetype ges the structured type from the property itself.

---
In reply to: [235493340](https://github.com/OData/odata.net/pull/1322#discussion_r235493340) [](ancestors = 235493340)